### PR TITLE
ADAPT-1771: Linting setup.

### PR DESCRIPTION
**Pull request by sherakama**
# READY FOR REVIEW

# Summary
- Sets and fixes linting issues 
- Uses prettier and airbnb standards
- Works in code climate

# Needed By (Date)
- Whenever but not too much longer as these changes will conflict at some point.

# Urgency
- 0

# Steps to Test

1. Review code changes, particularly around the eslint config
2. Review build site to ensure that it still functions and looks as it should

# Affected Projects or Products
- Give, Alumni

# Associated Issues and/or People
- ADAPT-1771

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Review by rebeccahongsf — APPROVED** [review]
Thanks for pulling this together @sherakama! Everything looks good to me 👍

File: src/DismissButton/DismissButton.stories.js [hunk 1]
```diff
@@ -1,90 +1,91 @@
...
+    },
+  },
 };
 
-const DismissButtonTemplate = ({ ...rest }) => {
-  return (
-    <DismissButton {...rest} />
-  );
-};
+const DismissButtonTemplate = ({ ...rest }) => <DismissButton {...rest} />;
```
Comment (yvonnetangsu) [inline src/DismissButton/DismissButton.stories.js:48]: Oh fancy one line returns

File: src/common/icon/getIconOption.js [hunk 2]
```diff
@@ -5,39 +5,40 @@
...
-      heroicon = 'mail';
+    case "jump":
+      heroicon = "chevron-down";
       break;
 
-    case 'jump':
-      heroicon = 'chevron-down';
+    case "action":
+    default:
+      heroicon = "chevron-right";
```
Comment (yvonnetangsu) [inline src/common/icon/getIconOption.js:41]: Looks like the forced default makes the "none" option stops working
https://deploy-preview-30--decanter-react.netlify.app/?path=/story/simple-cta-link--no-icon

Similarly for the "unset" size option for CTA Button
https://deploy-preview-30--decanter-react.netlify.app/?path=/story/simple-cta-link-button--solid&args=size:unset

File: src/CtaButton/CtaButton.stories.js [hunk 3]
```diff
@@ -1,181 +1,181 @@
...
-  link: 'https://stanford.edu',
+  variant: "unset",
+  size: "unset",
+  icon: "external",
+  animate: "top-right",
+  text: "With Custom Classes",
+  link: "https://stanford.edu",
   className:
-    'su-uppercase su-tracking-widest su-font-semibold su-bg-lagunita-dark hocus:su-bg-plum su-text-white hocus:su-text-white su-p-30',
+    "su-uppercase su-tracking-widest su-font-semibold su-bg-lagunita-dark hocus:su-bg-plum su-text-white hocus:su-text-white su-p-30",
```
Comment (yvonnetangsu) [inline src/CtaButton/CtaButton.stories.js:142]: Looks like the custom padding class `su-p-30` doesn't working anymore but it's not obvious why 🤔 

https://deploy-preview-30--decanter-react.netlify.app/?path=/story/simple-cta-link-button--custom
vs
https://decanter-react.netlify.app/?path=/story/simple-cta-link-button--custom

Comment (yvonnetangsu) [reply src/CtaButton/CtaButton.stories.js:142]: The padding class also stops working for the Button
https://deploy-preview-30--decanter-react.netlify.app/?path=/story/simple-button--custom

I just double checked Decanter 7 - that class should still be there.

**Review by yvonnetangsu — CHANGES_REQUESTED** [review]
Thanks for doing this, @sherakama! Just those comments on the added default cases causing some changes in variants, and the mysterious disappearing padding class in the stories wit custom styling.

File: src/CtaButton/CtaButton.stories.js (hunk 3 above)
Comment (sherakama) [reply src/CtaButton/CtaButton.stories.js:142]: There was a difference in classes being applied and it had to do with the default value in a switch statement. I made a couple of changes.

Screen Shot 2021-05-06 at 10 01 12 PM

File: src/common/icon/getIconOption.js (hunk 2 above)
Comment (sherakama) [reply src/common/icon/getIconOption.js]: Removed the new default.

**Comment by sherakama** [comment]
@yvonnetangsu Thanks for the review. Please have another pass.

**Review by yvonnetangsu — APPROVED** [review]
Thanks for the changes - all looking good. GTG!

Btw - do you think we could do releases for this repo too since Make a Gift is using this already?

**Pull request merged**